sdk/ai/azure-ai-generative/tests/evaluate/e2etests/test_evaluate_e2e.py

```diff
@@ -56,3 +56,96 @@ def test_evaluate_built_in_metrics(self, e2e_openai_api_base, e2e_openai_api_key
+            {"question": "How do you log a model?", "context": "Logging can be done using any OSS Sdk",
+             "answer": "There are a few ways to log models in Azure Machine Learning. \n\nOne way is to use the `register_model()` method of the `Run` object. The `register_model()` method logs a model file in the Azure Machine Learning service workspace and makes it available for deployment. Here's an example:\n\n```python\nfrom azureml.core import Model\n\nmodel_path = '.\/outputs\/my_model.pkl'\nmodel = Model.register(workspace=ws, model_path=model_path, model_name='my_model')\n```\n\nThis code registers the model file located at `model_path` to the Azure Machine Learning service workspace with the name `my_model`. \n\nAnother way to log a model is to save it as an output of a `Run`. If your model generation code is part of a script or Jupyter notebook that runs as an Azure Machine Learning experiment, you can save the model file as an output of the `Run` object. Here's an example:\n\n```python\nfrom sklearn.linear_model import LogisticRegression\nfrom azureml.core.run import Run\n\n# Initialize a run object\nrun = Run.get_context()\n\n# Train your model\nX_train, y_train = ...\nclf = LogisticRegression().fit(X_train, y_train)\n\n# Save the model to the Run object's outputs directory\nmodel_path = 'outputs\/model.pkl'\njoblib.dump(value=clf, filename=model_path)\n\n# Log the model as a run artifact\nrun.upload_file(name=model_path, path_or_stream=model_path)\n```\n\nIn this code, `Run.get_context()` retrieves the current run context object, which you can use to track metadata and metrics for the run. After training your model, you can use `joblib.dump()` to save the model to a file, and then log the file as an artifact of the run using `run.upload_file()`."},
+        ]
+
+        custom_metric_name = "custom_relevance"
+        from azure.ai.generative.evaluate.metrics import PromptMetric
+        custom_prompt_metric = PromptMetric.from_template(path="test_template.jinja2", name=custom_metric_name)
+
+        async def answer_length(*, data, **kwargs):
```

Comment:
I was kind of thinking the custom code metric function only accepting `**kwargs`. So that inside the function, it will be like `kwargs["question"]`, `kwargs["answer"]`. What do you think? The benefit of this is to remove one layer of nesting of the parameters.

Reply:
Idea here is to be explicit that data parameter has the data needed for metrics calculation. Wrapping it in kwargs makes it hard to understand.

Reply:
Which part do you think will be hard to understand? Here's what's more natural in my mind

Test data:
{"question": "...", "truth": "...."}
{"question": "...", "truth": "...."}


After invoking the target, we got
{"question": "...", "truth": "....", "context", "...", "answer": "...."}
{"question": "...", "truth": "....", "context", "...", "answer": "...."}

Then the custom code function should be able to accept `question`, `truth`, `question`, `context` directly. E.g.,

```python
def my_metric(*, question = None, truth = None, context = None, answer = None, **kwargs):
...
```

Wrapping up fields in another parameter called `data` seems unnecessary to me.

Reply:
Will consider this and discuss this internally.